bessctl/commands.py

```diff
@@ -1856,46 +1856,33 @@ def _capture_module(cli, module_name, direction, gate, opts, program, hook_fn):
     cli.fout.write('  Running: %s\n' % tcpdump_cmd)
     proc = subprocess.Popen(tcpdump_cmd, shell=True, preexec_fn=os.setsid)
 
-    cli.bess.pause_all()
+    unhook = True
     try:
         hook_fn(True, module_name, direction, gate, fifo)
-    except cli.bess.Error:
-        # kill all descendants in the process group
-        os.killpg(proc.pid, signal.SIGTERM)
-
-        try:
-            os.close(fd)
-            os.unlink(fifo)
-            os.system('stty sane')  # more/less may screw the terminal
-        except:
-            pass
-
-        raise
-
-    finally:
-        cli.bess.resume_all()
-
-    try:
         proc.wait()
     except KeyboardInterrupt:
         # kill all descendants in the process group
         os.killpg(proc.pid, signal.SIGTERM)
+    except cli.bess.Error:
+        unhook = False
+        # kill all descendants in the process group
+        os.killpg(proc.pid, signal.SIGTERM)
+        raise
```

Comment:
I'm not sure if this is a good or bad idea, but to collect zombies we should call `proc.wait()` after any `os.killpg(proc.pid, ...)`. The reason it might be a bad idea is that if the process refuses to die, `proc.wait` could wait forever.

Reply:
Right. umm.. At least, users can catch what is happening.